src/collapse/collapse-content.tsx

```diff
@@ -36,19 +35,36 @@ export const CollapseContent: React.FC<PropsWithChildren<Props>> = ({
   const contentRef = useRef<HTMLDivElement | null>(null);
   const initialContentHeight = useRef<number>(minHeight);
   const contentHeight = useRef<number>(DEFAULT_HEIGHT);
-  const [dimensions, setDimensions] = useState({
-    width: 0,
-    height: 0,
-  });
+
+  const [dimensions, setDimensions] = useState({width: 0, height: 0});
   const [height, setHeight] = useState<string>(toPx(minHeight));
-  const [showFade, setShowFade] = useState<boolean>(true);
+  const [showFade, setShowFade] = useState<boolean>(collapsed);
+
+  const [shouldHideContent, setShouldHideContent] = useState<boolean>(
+    collapsed && initialContentHeight.current <= DEFAULT_HEIGHT,
```

Comment:
Reading the ref value during rendering is not recommended, and it will not be compatible with React Compiler. It is better to do it differently right away.

https://react.dev/learn/referencing-values-with-refs#:~:text=rethink%20your%20approach.-,Don%E2%80%99t%20read%20or%20write%20ref.current%20during%20rendering,-.%20If%20some

Reply:
> I believe that since this defines the initial state for the useState hook and the rendering does not depend on this ref, it should not create any issues.

But `eslint-plugin-react-compiler` will report an error in this case if I understand correctly.